theano/tensor/tests/test_elemwise.py

```diff
@@ -870,6 +870,10 @@ def test_reduce_custom_dtype(self):
                         (topo, output_dtype)
                     data = numpy.random.rand(3, 4) * 10
                     data = data.astype(input_dtype)
+                    if output_dtype == 'float16' and method == 'prod':
+                        # We will likely get something infinite,
+                        # and DebugMode will complain.
+                        data = data[0:1]
```

Comment:
If we remove `* 10`, then all numbers will be cast to zero if `input_dtype` is an integer dtype.
With `* 2`, still half of it would be 0, which makes the test for product pretty meaningless.
That's why I opted for fewer numbers, but more varied (especially when cast as int).